java-frontend/src/main/java/org/sonar/java/model/JParser.java

```diff
@@ -1228,7 +1276,12 @@ private VariableTreeImpl convertVariable(SingleVariableDeclaration e) {
       );
     }
     t.variableBinding = e.resolveBinding();
-    declaration(t.variableBinding, t);
+    if (t.variableBinding != null) {
+      if (t.type() instanceof InferedTypeTree inferredType) {
+        inferredType.typeBinding = t.variableBinding.getType();
+      }
+      declaration(t.variableBinding, t);
```

Comment:
What do you think if we log a warming when there is no binding? Would it be too noisy?